components/script/dom/messageport.rs

```diff
@@ -148,7 +155,21 @@ impl MessagePort {
         }
 
         // Step 5
-        let data = structuredclone::write(cx, message, Some(transfer))?;
+        let data = if self.forbid_structured_clone {
+            let realm = enter_realm(self);
+            PostMessageData::Serialized(
+                jsval_to_webdriver(
+                    cx,
+                    &self.global(),
+                    message,
+                    InRealm::entered(&realm),
+                    CanGc::note(),
+                )
+                .map_err(|_| Error::InvalidState(None))?,
```

Comment:
So this means that from the script side, one can only send stuff that is compatible with `jsval_to_webdriver`, right?